packages/jellyfish-api-core/__tests__/category/loan/depositToVault.test.ts

```diff
@@ -0,0 +1,306 @@
+    })
+    await tGroup.get(0).generate(1)
+
+    vaultAddress = await tGroup.get(0).generateAddress()
+    vaultId = await tGroup.get(0).rpc.loan.createVault({
+      ownerAddress: vaultAddress,
+      loanSchemeId: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+
+    vaultId1 = await tGroup.get(0).rpc.loan.createVault({
+      ownerAddress: await tGroup.get(0).generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+
+    // set up liquidated vault here
+    liqVaultId = await tGroup.get(0).rpc.loan.createVault({
+      ownerAddress: await tGroup.get(0).generateAddress(),
+      loanSchemeId: 'scheme'
+    })
+    await tGroup.get(0).generate(1)
+
+    await tGroup.get(0).rpc.loan.depositToVault({
+      vaultId: liqVaultId, from: collateralAddress, amount: '10000@DFI'
+    })
+    await tGroup.get(0).generate(1)
+
+    await tGroup.get(0).rpc.loan.depositToVault({
+      vaultId: liqVaultId, from: collateralAddress, amount: '1@CAT'
+    })
+    await tGroup.get(0).generate(1)
+
+    await tGroup.get(0).rpc.loan.takeLoan({
+      vaultId: liqVaultId,
+      amounts: '100@TSLA'
+    })
+    await tGroup.get(0).generate(1)
+
+    // liquidated: true
+    await tGroup.get(0).rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '100000@TSLA', currency: 'USD' }] })
+    await tGroup.get(0).generate(1)
+    await tGroup.waitForSync()
+  }
+
+  it('should be failed as first deposit must be DFI', async () => {
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      vaultId: vaultId, from: collateralAddress, amount: '1@BTC'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow('At least 50% of the vault must be in DFI thus first deposit must be DFI')
+  })
+
+  it('should be failed as insufficient fund', async () => {
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      vaultId: vaultId, from: collateralAddress, amount: '99999@DFI'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Insufficient funds: can't subtract balance of ${collateralAddress}: amount 20000.00000000 is less than 99999.00000000`)
+  })
+
+  it('should be failed as different auth address', async () => {
+    const addr = await tGroup.get(1).generateAddress()
+    const promise = tGroup.get(0).rpc.loan.depositToVault({
+      vaultId: vaultId, from: addr, amount: '1@DFI'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Incorrect authorization for ${addr}`)
+  })
+
+  it('should be failed as deposit by other node', async () => {
+    const promise = tGroup.get(1).rpc.loan.depositToVault({
+      vaultId: vaultId, from: collateralAddress, amount: '300@DFI'
+    })
+    await expect(promise).rejects.toThrow(RpcApiError)
+    await expect(promise).rejects.toThrow(`Incorrect authorization for ${collateralAddress}`)
+  })
```

Comment:
@canonbrother Regarding https://github.com/DeFiCh/jellyfish/pull/661/files#r711709439,  actually this is what I wanted to add.
```suggestion
it('should be able to depositToVault by anyone', async () => {
    const vaultBefore = await tGroup.get(0).container.call('getvault', [vaultId])
    
    const addr = await tGroup.get(1).generateAddress()
    await tGroup.get(1).token.dfi({ address: addr, amount: 100 })
    await tGroup.get(1).generate(1)
    await tGroup.waitForSync()

    const depositId = await tGroup.get(1).rpc.loan.depositToVault({
      vaultId: vaultId, from: addr, amount: '2@DFI'
    })
    expect(typeof depositId).toStrictEqual('string')
    await tGroup.get(1).generate(1)
    await tGroup.waitForSync()

    const vaultAfter = await tGroup.get(0).container.call('getvault', [vaultId])

    //Check the values vaultBefore, vaultAfter here.

  })
```

Reply:
yep.. added this and removed 1 `diff auth` test